.github/workflows/release.yml

```diff
@@ -167,21 +157,17 @@ jobs:
           path: artifacts
 
       - name: Set release upload URL and release version
-        shell: bash
         run: |
-          release_upload_url="$(cat artifacts/release-upload-url)"
-          echo "RELEASE_UPLOAD_URL=$release_upload_url" >> $GITHUB_ENV
-          echo "release upload url: $RELEASE_UPLOAD_URL"
-          release_version="$(cat artifacts/release-version)"
-          echo "RELEASE_VERSION=$release_version" >> $GITHUB_ENV
-          echo "release version: $RELEASE_VERSION"
+          echo "UPLOAD_URL=$(< artifacts/release-upload-url)" >> "$GITHUB_ENV"
```

Comment:
I didn't know this syntax existed! Initializing variables from file content, neat!

Reply:
`$(< artifacts/release-upload-url)` actually has the same effect as `$(cat artifacts/release-upload-url)`, just without using `cat` or any other subprocess--the shell opens and reads the file. The Bash reference manual [documents it](https://www.gnu.org/software/bash/manual/bash.html#Command-Substitution) as a special form of command substitution, though its semantics entail that no command is run. A copy of [the manual for bash 3.2](https://www.bashcookbook.com/bashinfo/source/bash-3.2/doc/bashref.pdf) confirms that it was present back then too, and thus safe to use even in `bash` on macOS.

Normally, an unquoted `<` at the start of a command is just input redirection, the same as any other command. But when it is inside `$(` `)` and the entire "command" is a path, `bash` treats it specially. This `bash` feature is not standardized. However, to fix the way environment variables are set and used, I had already made all script steps use `bash` on all operating systems, so it is definitely available. (I wouldn't use it in a command passed to `docker run`, though, since the shell that uses is often `sh`.)

From a technical perspective, replacing `cat` with `<` is independent of the other change here, which was to remove the `echo` commands that tried to print the effect of setting the environment variable. These commands never worked, because appending to the file that `$GITHUB_ENV` expands to is setting environment variables for subsequent steps, not the current step, not even subsequent commands of the script run in the current step. Since in this case the variables didn't seem to require special attention and also are already available for inspection in the workflow run, I just removed them.

Removing those other `echo` commands allowed the initial assignment be eliminated while still avoiding duplication, and thus allowed the variables to be set with one command each. I felt that changing `$(cat ...)` to `$(< ...)` was part of keeping that new combined command as short and simple as it reasonably could be, which is why I did it at the same time.